src/Symfony/Bridge/Monolog/Processor/AbstractTokenProcessor.php

```diff
@@ -42,10 +42,15 @@ public function __invoke(array $record): array
 
         if (null !== $token = $this->getToken()) {
             $record['extra'][$this->getKey()] = [
-                'username' => $token->getUsername(),
                 'authenticated' => $token->isAuthenticated(),
                 'roles' => $token->getRoleNames(),
             ];
+
+            if (method_exists($token, 'getUserIdentifier')) {
+                $record['extra'][$this->getKey()]['username'] = $record['extra'][$this->getKey()]['user_identifier'] = $token->getUserIdentifier();
```

Comment:
We don't want to keep adding `username` to the context for up-to-date implementations, do we?

Reply:
I'm not sure. E.g. the `LdapUser` and `User` are up-to-date, so that would mean upgrading to 5.3 will automatically change your log context. At my work, the log processing is flexible enough to cope with this, but I'm not sure if this applies to everyone.